### PR TITLE
[#234] Additional test tags to exclude tests with known issues

### PR DESCRIPTION
**Pull request by ablatov**
- timeout added for EDI tests if they stuck because of bugs.
- tags functionality for robot tags increased (Input tags: "core,-edi", if "-" in tag name then exclude it).
- more suitable keyword used in EDI tests for running legionctl commands.

### CODE REVIEW

File: deploy/legionPipeline.groovy [hunk 1]
```diff
@@ -93,9 +93,19 @@ def createjenkinsJobs(String commitID) {
...
     withAWS(credentials: 'kops') {
     	withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
-            env.tags=tags.toString().trim()
-            env.robot_tags= !env.tags.empty ? "-i " + env.tags.replaceAll(',',' -i ') : ""
-            env.nose_tags = !env.tags.empty ? "-a " + env.tags.replaceAll(',',' -a ') : ""
+            def tags_list=tags.toString().split(',')
+            def robot_tags= []
+            def nose_tags = []
+            for (item in tags_list) {
+                if (item.contains('-')) robot_tags.add(item)
```
Comment (kirillmakhonin) [inline deploy/legionPipeline.groovy]: It looks hard to use

File: legion/legion/utils.py [hunk 2]
```diff
@@ -487,7 +487,7 @@ def send_header_to_stderr(header, value):
     :type value: str
     :return: None
     """
-    message = '{}{}:{}'.format(legion.containers.headers.STDERR_PREFIX, header, value)
+    message = '\n{}{}:{}'.format(legion.containers.headers.STDERR_PREFIX, header, value)
```
Comment (kirillmakhonin) [inline legion/legion/utils.py]: This is not relative to issue

File: tests/robot/4_check_edi.robot [hunk 3]
```diff
@@ -1,10 +1,12 @@
 *** Settings ***
 Documentation       Legion's EDI operational check
+Test Timeout        5 minutes
```
Comment (kirillmakhonin) [inline tests/robot/4_check_edi.robot]: Is 5 minutes enough?

File: deploy/legionPipeline.groovy (hunk 1 above)
Comment (aliaksandr-d) [reply deploy/legionPipeline.groovy]: Same here. Need to think over tags_list format.

File: tests/robot/4_check_edi.robot (hunk 3 above)
Comment (ablatov) [reply tests/robot/4_check_edi.robot]: @kirillmakhonin more than enough, will configure more closely when it starts work.

**Comment by ablatov** [comment]
@kirillmakhonin @aliaksandr-d look up for changes, please.
Input tags: core,-edi
if "-" in tag name then exclude it.

File: deploy/legionPipeline.groovy [hunk 4]
```diff
@@ -93,9 +93,23 @@ def createjenkinsJobs(String commitID) {
...
     withAWS(credentials: 'kops') {
     	withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
-            env.tags=tags.toString().trim()
-            env.robot_tags= !env.tags.empty ? "-i " + env.tags.replaceAll(',',' -i ') : ""
-            env.nose_tags = !env.tags.empty ? "-a " + env.tags.replaceAll(',',' -a ') : ""
+            def tags_list=tags.toString().trim().split(',')
+            def robot_tags= []
+            def nose_tags = []
+            for (item in tags_list) {
+                if (item.contains('-')) {
```
Comment (kirillmakhonin) [inline deploy/legionPipeline.groovy]: `item.contains` -> `item.startsWith()`?

File: deploy/legionPipeline.groovy [hunk 5]
```diff
@@ -93,9 +93,23 @@ def createjenkinsJobs(String commitID) {
...
+                    item = item.replaceAll("-","")
```
Comment (kirillmakhonin) [inline deploy/legionPipeline.groovy]: `replaceAll` - bad

File: deploy/legionPipeline.groovy (hunk 4 above)
Comment (kirillmakhonin) [reply deploy/legionPipeline.groovy]: Maybe it would be better to use boolean variable with appropriate name?

File: deploy/legionPipeline.groovy [hunk 6]
```diff
@@ -93,9 +93,23 @@ def createjenkinsJobs(String commitID) {
...
     	withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
-            env.tags=tags.toString().trim()
-            env.robot_tags= !env.tags.empty ? "-i " + env.tags.replaceAll(',',' -i ') : ""
-            env.nose_tags = !env.tags.empty ? "-a " + env.tags.replaceAll(',',' -a ') : ""
+            def tags_list=tags.toString().trim().split(',')
+            def robot_tags= []
+            def nose_tags = []
+            for (item in tags_list) {
+                if (item.startsWith('-')) {
+                    item = item.replace("-","")
```
Comment (kirillmakhonin) [inline deploy/legionPipeline.groovy]: This replaces all occurrences of `-` in tag. Is that good?

Comment (ablatov) [reply deploy/legionPipeline.groovy]: @kirillmakhonin that is exactly what we want here.

**Pull request merged**